### PR TITLE
Fix/9484 no submit while collect in background

### PR DESCRIPTION
**Pull request by marcussc**
## Description
We don't want to submit PPA every time we collect PPA data in the background since we have a dedicated PPA submission task in the background.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9484

### CODE REVIEW

**Review by nickguendling — APPROVED** [review]

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/PPAnalyticsCollector/PPAnalyticsCollector.swift [hunk 1]
```diff
@@ -59,8 +59,13 @@ enum PPAnalyticsCollector {
 			Analytics.logSubmissionMetadata(submissionMetadata)
 		}
 
-		// At the end, try to submit the data. In the submitter are all the checks that we do not submit the data to often.
-		Analytics.triggerAnalyticsSubmission()
+		// Only trigger the submission if the app is in foreground (Exposure-9484)
+		if UIApplication.shared.applicationState == .active {
+			// At the end, try to submit the data. In the submitter are all the checks that we do not submit the data to often.
+			Log.info("Triggering submission after collection new PPA data", log: .ppa)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/PPAnalyticsCollector/PPAnalyticsCollector.swift:65]:
```suggestion
			Log.info("Triggering submission after collecting new PPA data", log: .ppa)
```

File: src/xcode/ENA/ENA/Source/Services/PPAnalyticsCollector/PPAnalyticsCollector.swift [hunk 2]
```diff
@@ -59,8 +59,13 @@ enum PPAnalyticsCollector {
 			Analytics.logSubmissionMetadata(submissionMetadata)
 		}
 
-		// At the end, try to submit the data. In the submitter are all the checks that we do not submit the data to often.
-		Analytics.triggerAnalyticsSubmission()
+		// Only trigger the submission if the app is in foreground (Exposure-9484)
+		if UIApplication.shared.applicationState == .active {
+			// At the end, try to submit the data. In the submitter are all the checks that we do not submit the data to often.
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/PPAnalyticsCollector/PPAnalyticsCollector.swift:64]:
```suggestion
			// At the end, try to submit the data. In the submitter are all the checks that we do not submit the data too often.
```

File: src/xcode/ENA/ENA/Source/Services/PPAnalyticsCollector/PPAnalyticsCollector.swift [hunk 3]
```diff
@@ -59,8 +59,13 @@ enum PPAnalyticsCollector {
 			Analytics.logSubmissionMetadata(submissionMetadata)
 		}
 
-		// At the end, try to submit the data. In the submitter are all the checks that we do not submit the data to often.
-		Analytics.triggerAnalyticsSubmission()
+		// Only trigger the submission if the app is in foreground (Exposure-9484)
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Services/PPAnalyticsCollector/PPAnalyticsCollector.swift:62]:
```suggestion
		// Only trigger the submission if the app is in foreground (Exposure-9484). For background, the submission is triggered explicit.
```

**Review by pascal-brause — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/PPAnalyticsCollector/PPAnalyticsCollector.swift (hunk 3 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Services/PPAnalyticsCollector/PPAnalyticsCollector.swift:62]: explicitly*

**Pull request merged**